internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,671 @@
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	// agentServiceConfigName is the one and only name for an AgentServiceConfig
+	// supported in the cluster. Any others will be ignored.
+	agentServiceConfigName = "agent"
+	// filesystemPVCName is the name of the PVC created for assisted-service's filesystem.
+	filesystemPVCName = "agent-filesystem"
+	// databasePVCName is the name of the PVC created for postgresql.
+	databasePVCName = "agent-database"
+
+	name                                = "assisted-service"
+	databaseName                        = "postgres"
+	databaseSecretName                  = databaseName
+	databasePort                  int32 = 5432
+	servicePort                   int32 = 8090
+	assistedServiceDeploymentName       = "assisted-service"
+
+	// assistedServiceContainerName is the Name property of the assisted-service container
+	assistedServiceContainerName string = "assisted-service"
+	// databaseContainerName is the Name property of the postgres container
+	databaseContainerName string = databaseName
+)
+
+// AgentServiceConfigReconciler reconciles a AgentServiceConfig object
+type AgentServiceConfigReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	// Namespace the operator is running in
+	Namespace string
+}
+
+// +kubebuilder:rbac:groups=adi.io.my.domain,resources=agentserviceconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=adi.io.my.domain,resources=agentserviceconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=adi.io.my.domain,resources=agentserviceconfigs/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;list;watch;create;update;patch;delete
+
+func (r *AgentServiceConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	_ = r.Log.WithValues("agentserviceconfig", req.NamespacedName)
```

Comment:
No. This adds the key/value pair to the context of the `r.Log` logger.

Reply:
Hm, so this would be a problem if we set `MaxConcurrentReconciles` > 1?

Reply:
Trying to find the concrete implementation of this is nearly a nightmare. I think it [ends up here](https://github.com/uber-go/zap/blob/c0dfce3/zapcore/core.go#L72-L76). Suffice it to say, I would be very surprised if this adds values to `r.Log`. Do you have a reference that says it will? Or are you seeing it actually do that?

Reply:
you are right.